### ISSUE TITLE
TODO LIST

### ISSUE THREAD

**Issue by kimocoder** [offensive-security/nethunter-app#213]
Ok people, we still got some work to do..

Here's the TODO list for now. @Re4son will add more.

* We got two busyboxes running
  - One with v1.25.0-NetHunter
  - One (cmd: busybox) with v1.30.1 (when in chroot)
  - busybox could be linked or updated,

* "killkali" has some mounting issues (will add picture)
* "bootkali" has 1 output (issue) (will add picture)

* The UI could be improved (material design?)
* We could simply add more features, like network tools in menu (output from our tools in kali)
  ping, trace, whois, traceroute.. whatever
* We should rebuild the "Deauther" and add for scanning and attacking using our tools,,

**Comment by kimocoder** [comment]
**API**
Android Network API link [here](https://developer.android.com/training/basics/network-ops)

**Comment by Re4son** [comment]
Great list @kimocoder . Might be easier to have individual issues for each of them. That would make it easier to work on them individually.
1. Busyboxe: I have one and the same in all of my four different nethunter devices. Best to raise a separate issue because that looks like a bug in the image you're using.

2. The mount errors are caused by non existing filesystems:
a) /sdcard/kali-chroot.img: alternative chroot.img will be mountet if exists but usually doesn't
b) /system/lib/modules - where are your modules installed?
c) Does angler have an external sd-card?


Could you do me a favor please and try the nethunter.apk pre 2019.1 to see if those error where introduced by those massive commits over the last two months?

**Comment by kimocoder** [comment]
Nexus 6P on latest official firmware, Android 8 with WiFi modules, HID, CDROM, dwc3 patch with ACA (Y-cable support) patches. works great, but yeah we could do much more.

![Screenshot_20190419-084150](https://user-images.githubusercontent.com/4252297/56424981-61f08780-6280-11e9-8a75-a2d2553add27.png)

**Comment by Re4son** [comment]
Did you do a complete re-image before you installed the latest nethunter image? With all the testing you might have left over busyboxes from previous installs

**Comment by kimocoder** [comment]
Yepp, that's my mission this easter. I'll get them.

**Comment by Re4son** [comment]
Get what?

**Comment by kimocoder** [comment]
> Did you do a complete re-image before you installed the latest nethunter image? With all the testing you might have left over busyboxes from previous installs

Yes, everythime.

> Get what?

Get those mounts for away.

**Comment by Re4son** [comment]
Could you please check for busybox after you reflashed and rooted?
Then again after you installed nethunter?

Then install the pre-2019.1 nethunter app and check the errors during bootkali and killkali.

Thanks mate!

**Comment by kimocoder** [comment]
Yes, needs to be tomorrow, been reading like alot the past data :+1:

**Comment by kimocoder** [comment]
cSploit is way outdated the one MITM working is "Kill connections"
Therefor i ein zANTI2 instead.

**Comment by kimocoder** [comment]
And some menu where modules (kmod) where we could load modules and set them in startup. Something like this

![Screenshot_20190419-092346](https://user-images.githubusercontent.com/4252297/56426125-0ffe3080-6285-11e9-8afe-213e695e07f3.png)
